product_merge/wizards/product_merge_wizard.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib.openupgrade_merge_records import merge_records
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductMergeWizard(models.TransientModel):
+    _name = "product.merge.wizard"
+    _description = "Merge Products Wizard"
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product Model",
+        domain="[('id', 'in', product_ids)]",
+        required=True,
+    )
+    product_ids = fields.Many2many(
+        comodel_name="product.template", string="Products to Merge", required=True
+    )
+    attribute_ids = fields.Many2many(
+        comodel_name="product.attribute", string="Attributes", required=True
+    )
+    line_ids = fields.One2many(
+        comodel_name="product.merge.wizard.line",
+        inverse_name="wizard_id",
+        string="Attribute Mapping",
+        compute="_compute_line_ids",
+        store=True,
+        readonly=False,
+    )
+
+    def _check_minimum_products(self):
+        """
+        Ensure that at least two products are selected for merging.
+        """
+        for wizard in self:
+            if len(wizard.product_ids) < 2:
```

Comment:
An alert should be great on wizard form in order to warn user before clicking on merge.

Reply:
This check can't be done before the user completes their input
otherwise, they would receive a warning after adding the first product
IMO, it makes sense to trigger it when they click on the button